core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -338,6 +346,148 @@ private JarBuildItem buildThinJar(CurateOutcomeBuildItem curateOutcomeBuildItem,
         return new JarBuildItem(runnerJar, null, libDir);
     }
 
+    private JarBuildItem buildThinJar(CurateOutcomeBuildItem curateOutcomeBuildItem,
+            OutputTargetBuildItem outputTargetBuildItem,
+            TransformedClassesBuildItem transformedClasses,
+            ApplicationArchivesBuildItem applicationArchivesBuildItem,
+            PackageConfig packageConfig,
+            ApplicationInfoBuildItem applicationInfo,
+            List<GeneratedClassBuildItem> generatedClasses,
+            List<GeneratedResourceBuildItem> generatedResources,
+            MainClassBuildItem mainClassBuildItem) throws Exception {
+
+        Path buildDir = outputTargetBuildItem.getOutputDirectory()
+                .resolve(outputTargetBuildItem.getBaseName());
+        IoUtils.recursiveDelete(buildDir);
+        Files.createDirectories(buildDir);
+
+        //unmodified 3rd party dependencies
+        Path libDir = buildDir.resolve("lib");
+        Files.createDirectories(libDir);
+        //parent first entries
+        Path baseLib = buildDir.resolve("base-lib");
```

Comment:
Not really important, but maybe `base-lib` should be called `boot-lib` to better reflect what its about? `base-lib` sounds a little to generic